02-microarray/pathway-analysis_microarray_02_ora_with_webgestaltr.Rmd

```diff
@@ -0,0 +1,263 @@
+We suggest saving plots and results to `plots/` and `results/` directories, respectively, as these are automatically created by the notebook.
+From here you can customize this analysis example to fit your own scientific questions and preferences. 
+
+***
+
+<!-- Do not delete this line --> <a name="analysis" style="padding-top:56px;margin-top:-56px;">&nbsp;</a>
+
+## Over-Representation Analysis with `WebGestaltR`
+
+## Install libraries
+
+See our Getting Started page with [instructions for package installation](https://alexslemonade.github.io/refinebio-examples/01-getting-started/getting-started.html#what-you-need-to-install) for a list of the other software you will need, as well as more tips and resources.
+
+In this analysis, we will be using `WebGestaltR` [@Wang2020].
+
+```{r}
+if (!("WebGestaltR" %in% installed.packages())) {
+  # Install this package if it isn't installed yet
+  install.packages("WebGestaltR")
+}
+```
+
+Attach the packages we need for this analysis.
+
+```{r}
+# Attach the library
+library(WebGestaltR)
+
+# We will need this so we can use the pipe: %>%
+library(magrittr)
+```
+
+### Import data 
+
+We will read in the differential expression results we will download from online.
+These results are from a four group comparison we completed in the [microarray differential expression analysis for several groups](https://alexslemonade.github.io/refinebio-examples/02-microarray/differential-expression_microarray_02_several-groups.html) using [`limma`](https://bioconductor.org/packages/release/bioc/html/limma.html) [@Ritchie2015].
+The table contains Ensembl gene IDs, log fold-change, and adjusted p-values (FDR in this case).
+We can identify upregulated genes by filtering these results and use this list
+as input to ORA.
+
+```{r}
+# Define the url to your differential expression results file
+dge_url <- "https://refinebio-examples.s3.us-east-2.amazonaws.com/02-microarray/results/GSE37418/GSE37418_limma_results.tsv"
+# Replace with a link to your desired differential expression results TSV file or the file path to differential expression results TSV file on your local computer
+```
+
+Read in the file that is online with the differential expression results. 
+
+```{r}
+# Read in the contents of your differential expression results file
+dge_df <- readr::read_tsv(dge_url)
+```
+
+`read_tsv()` can read TSV files online and doesn't necessarily require you download it first. 
+However, should you decide you want to download the file to your `data/` directory, you can use `download.file(dge_url, destfile = file.path(data_dir, "GSE37418_limma_results.tsv"))`.
+
+Let's take a look at what these differential expression results look like. 
+
+```{r}
+dge_df
+```
+
+Here we'll use the `SHH` groups' log fold-change > 2 and FDR < 0.05 as cutoffs for determining 
+what genes are of interest.
+
+```{r}
+upregulated_genes <- dge_df %>%
+  dplyr::filter(SHHvsOther > 2, adj.P.Val < 0.05) %>%
```

Comment:
I feel like we might want to use the contrast p value here instead of the overall p value? Is this correct?

Reply:
I think it makes sense to show users in the `differential-expression_microarray_02_several-groups.Rmd` how to save the contrast p values to a file too. (We already extracted them anyway).